_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,649 @@
+
+```json
+POST /hotels-index/_search
+{
+    "size": 3,
+    "query": {
+        "knn": {
+            "location": {
+                "vector": [
+                    5.0,
+                    4.0
+                ],
+                "k": 3,
+                "filter": {
+                    "bool": {
+                        "must": [
+                            {
+                                "range": {
+                                    "rating": {
+                                        "gte": 8,
+                                        "lte": 10
+                                    }
+                                }
+                            },
+                            {
+                                "term": {
+                                    "parking": "true"
+                                }
+                            }
+                        ]
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+
+#### Sample Response
+
+The following response indicates that only three hotels met the filter criteria:
+
+
+```json
+{
+  "took" : 47,
+  "timed_out" : false,
+  "_shards" : {
+    "total" : 1,
+    "successful" : 1,
+    "skipped" : 0,
+    "failed" : 0
+  },
+  "hits" : {
+    "total" : {
+      "value" : 3,
+      "relation" : "eq"
+    },
+    "max_score" : 0.72992706,
+    "hits" : [
+      {
+        "_index" : "hotels-index",
+        "_id" : "3",
+        "_score" : 0.72992706,
+        "_source" : {
+          "location" : [
+            4.9,
+            3.4
+          ],
+          "parking" : "true",
+          "rating" : 9
+        }
+      },
+      {
+        "_index" : "hotels-index",
+        "_id" : "6",
+        "_score" : 0.3012048,
+        "_source" : {
+          "location" : [
+            6.4,
+            3.4
+          ],
+          "parking" : "true",
+          "rating" : 9
+        }
+      },
+      {
+        "_index" : "hotels-index",
+        "_id" : "5",
+        "_score" : 0.24154587,
+        "_source" : {
+          "location" : [
+            3.3,
+            4.5
+          ],
+          "parking" : "true",
+          "rating" : 8
+        }
+      }
+    ]
+  }
+}
+
+```
+
+## Additional complex filter query
+
+Depending on how restrictive you want your filter to be, you can add multiple query types to a single request, such as `term`, `wildcard`, `regexp`, and `range`. You can then filter out the search results with the Boolean clauses `must`, `should`, and `must_not`.
+
+#### Sample request
+
+The following request returns hotels that provide parking. This request illustrates alternate mechanisms to obtain the parking filter criteria. It uses a regular expression for the value `true`, a term query for the key-value pair `"parking":"true"`, a wildcard for the characters that spell "true", and the `must_not` clause to eliminate hotels with "parking" set to `false`.
```

Comment:
```suggestion
The following request returns hotels that provide parking. This request illustrates alternative mechanisms for obtaining the parking filter criteria. It uses a regular expression for the value `true`, a term query for the key-value pair `"parking":"true"`, a wildcard for the characters that spell "true", and the `must_not` clause to eliminate hotels with "parking" set to `false`:
```

Reply:
Are we illustrating multiple alternative mechanisms or just one? If the latter, "This request illustrates an alternative mechanism..."